core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1022,6 +1065,15 @@ public Builder removeBranch(String branch) {
       return this;
     }
 
+    public Builder removeTag(String tag) {
```

Comment:
Since these are so similar and we're calling them only after checking the type, I think it makes sense to have removeRef and setRef rather than separate methods by type.

Reply:
Sounds good, I think I had this before, but went back to having it separate. But yes, the logic is basically the same  so we can have it in a common method.